packages/experience/src/components/InputFields/InputField/index.tsx

```diff
@@ -47,33 +56,29 @@ const InputField = (
 
   const [isFocused, setIsFocused] = useState(false);
   const [hasValue, setHasValue] = useState(false);
-  const hasContent =
-    Boolean(isPrefixVisible) ||
-    hasValue ||
-    // Handle the case when this filed have a default value
-    Boolean(value);
-
-  const isActive = hasContent || isFocused;
 
   useEffect(() => {
-    const inputDom = innerRef.current;
-    if (!inputDom) {
-      return;
-    }
-
     /**
-     * Use a timeout to check if the input field has autofill value.
-     * Fix the issue that the input field doesn't have the active style when the autofill value is set.
-     * see https://github.com/facebook/react/issues/1159#issuecomment-1025423604
+     * Should listen to the value prop to update the hasValue state.
      */
-    const checkAutoFillTimeout = setTimeout(() => {
-      setHasValue(inputDom.matches('*:-webkit-autofill'));
-    }, 200);
+    setHasValue(Boolean(value));
+  }, [value]);
+
+  /**
+   * Fix the issue that the input field doesn't have the active style when the autofill value is set.
+   * Modern browsers will trigger an animation event when the input field is autofilled.
+   */
+  const handleAnimationStart = useCallback((event: AnimationEvent<HTMLInputElement>) => {
+    if (event.animationName === 'onautofillstart') {
+      setHasValue(true);
+    }
+  }, []);
+
+  const isActive = useMemo(() => {
+    const hasContent = Boolean(isPrefixVisible) || hasValue;
 
-    return () => {
-      clearTimeout(checkAutoFillTimeout);
-    };
-  }, [innerRef]);
+    return hasContent || isFocused;
+  }, [hasValue, isFocused, isPrefixVisible]);
```

Comment:
This simple boolean OR operation does not deserve a `useMemo` IMO

Reply:
updated.